diff -Naur Python-2.7.13/setup.py Python-2.7.13.tpg/setup.py
--- Python-2.7.13/setup.py	2017-02-21 18:55:19.000000000 +0000
+++ Python-2.7.13.tpg/setup.py	2017-02-21 18:57:16.334502283 +0000
@@ -2153,7 +2153,7 @@
             with open(ffi_h) as f:
                 for line in f:
                     line = line.strip()
-                    if line.startswith(('#define LIBFFI_H',
+                    if line.startswith(('#define _MULTIARCH_HEADER',
                                         '#define ffi_wrapper_h')):
                         break
                 else:
