--- Python-2.7.1/setup.py.db5~	2011-04-10 05:08:02.755998361 +0200
+++ Python-2.7.1/setup.py	2011-04-10 05:23:59.644035612 +0200
@@ -809,6 +813,9 @@ class PyBuildExt(build_ext):
         # construct a list of paths to look for the header file in on
         # top of the normal inc_dirs.
         db_inc_paths = [
+            '/usr/include/db5',
+            '/usr/local/include/db5',
+            '/opt/sfw/include/db5',
             '/usr/include/db4',
             '/usr/local/include/db4',
             '/opt/sfw/include/db4',
@@ -819,6 +826,16 @@ class PyBuildExt(build_ext):
             '/sw/include/db4',
             '/sw/include/db3',
         ]
+        # 5.x minor number specific paths
+        for x in gen_db_minor_ver_nums(5):
+            db_inc_paths.append('/usr/include/db5%d' % x)
+            db_inc_paths.append('/usr/include/db5.%d' % x)
+            db_inc_paths.append('/usr/local/BerkeleyDB.5.%d/include' % x)
+            db_inc_paths.append('/usr/local/include/db5%d' % x)
+            db_inc_paths.append('/pkg/db-5.%d/include' % x)
+            db_inc_paths.append('/opt/db-5.%d/include' % x)
+            # MacPorts default (http://www.macports.org/)
+            db_inc_paths.append('/opt/local/include/db5%d' % x)
         # 4.x minor number specific paths
         for x in gen_db_minor_ver_nums(4):
             db_inc_paths.append('/usr/include/db4%d' % x)
@@ -845,6 +862,10 @@ class PyBuildExt(build_ext):
         for dn in inc_dirs:
             std_variants.append(os.path.join(dn, 'db3'))
             std_variants.append(os.path.join(dn, 'db4'))
+            std_variants.append(os.path.join(dn, 'db5'))
+            for x in gen_db_minor_ver_nums(5):
+                std_variants.append(os.path.join(dn, "db5%d"%x))
+                std_variants.append(os.path.join(dn, "db5.%d"%x))
             for x in gen_db_minor_ver_nums(4):
                 std_variants.append(os.path.join(dn, "db4%d"%x))
                 std_variants.append(os.path.join(dn, "db4.%d"%x))
