diff -Naur Python-2.7.14/setup.py Python-2.7.14.tpg/setup.py
--- Python-2.7.14/setup.py	2017-10-27 09:57:40.780559000 +0000
+++ Python-2.7.14.tpg/setup.py	2017-10-27 10:04:34.016759458 +0000
@@ -2095,7 +2095,7 @@
         return True
 
     def detect_ctypes(self, inc_dirs, lib_dirs):
-        self.use_system_libffi = False
+        self.use_system_libffi = ('--with-system-ffi' in sysconfig.get_config_var("CONFIG_ARGS"))
         include_dirs = []
         extra_compile_args = []
         extra_link_args = []
@@ -2140,7 +2140,7 @@
                              sources=['_ctypes/_ctypes_test.c'])
         self.extensions.extend([ext, ext_test])
 
-        if not '--with-system-ffi' in sysconfig.get_config_var("CONFIG_ARGS"):
+        if not self.use_system_libffi:
             return
 
         if host_platform == 'darwin':
@@ -2165,15 +2165,15 @@
                           'ffi_wrapper_h'.format(ffi_h))
         ffi_lib = None
         if ffi_inc is not None:
-            for lib_name in ('ffi_convenience', 'ffi_pic', 'ffi'):
+            for lib_name in ('ffi', 'ffi_convenience', 'ffi_pic', 'ffi'):
                 if (self.compiler.find_library_file(lib_dirs, lib_name)):
                     ffi_lib = lib_name
                     break
 
-        if ffi_inc and ffi_lib:
+        if ffi_inc:
             ext.include_dirs.extend(ffi_inc)
+        if ffi_lib:
             ext.libraries.append(ffi_lib)
-            self.use_system_libffi = True
 
 
 class PyBuildInstall(install):
