--- setup.py.orig	2013-04-08 16:19:07.630643642 +0400
+++ setup.py	2013-04-08 16:18:53.265642396 +0400
@@ -688,7 +688,7 @@
         # Operations on audio samples
         # According to #993173, this one should actually work fine on
         # 64-bit platforms.
-        exts.append( Extension('audioop', ['audioop.c']) )
+        exts.append( Extension('audioop', ['audioop.c'], libraries=math_libs) )
 
         # Disabled on 64-bit platforms
         if sys.maxint != 9223372036854775807L:
@@ -1518,7 +1518,7 @@
             # This requires sizeof(int) == sizeof(long) == sizeof(char*)
             dl_inc = find_file('dlfcn.h', [], inc_dirs)
             if (dl_inc is not None) and (host_platform not in ['atheos']):
-                exts.append( Extension('dl', ['dlmodule.c']) )
+                exts.append( Extension('dl', ['dlmodule.c'], libraries = ['dl'] ) )
             else:
                 missing.append('dl')
         else:
@@ -2038,10 +2038,11 @@
                         include_dirs=include_dirs,
                         extra_compile_args=extra_compile_args,
                         extra_link_args=extra_link_args,
-                        libraries=[],
+                        libraries=['m', 'dl'],
                         sources=sources,
                         depends=depends)
         ext_test = Extension('_ctypes_test',
+			     libraries=['m'],
                              sources=['_ctypes/_ctypes_test.c'])
         self.extensions.extend([ext, ext_test])
 
